src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/PageFaultMetricsIT.java

```diff
@@ -0,0 +1,183 @@
+    }
+  }
+
+  @Test
+  public void checkPaging_MinfltRate() throws Exception {
+    //read metric from local node
+    List<JsonResponseNode> responseNodeList =
+        readMetric(PERFORMANCE_ANALYZER_BASE_ENDPOINT + "/metrics/?metrics=Paging_MinfltRate&agg=max");
+    Assert.assertEquals(1, responseNodeList.size());
+    validateMinorPageFaultMetric(responseNodeList.get(0));
+
+    //read metric from all nodes in cluster
+    responseNodeList =
+        readMetric(PERFORMANCE_ANALYZER_BASE_ENDPOINT + "/metrics/?metrics=Paging_MinfltRate&agg=max&nodes=all");
+    int nodeNum = nodeIDs.size();
+    Assert.assertEquals(nodeNum, responseNodeList.size());
+    for (int i = 0; i < nodeNum; i++) {
+      validateMinorPageFaultMetric(responseNodeList.get(i));
+    }
+  }
+
+  @Test
+  public void checkPaging_RSS() throws Exception {
+    //read metric from local node
+    List<JsonResponseNode> responseNodeList =
+        readMetric(PERFORMANCE_ANALYZER_BASE_ENDPOINT + "/metrics/?metrics=Paging_RSS&agg=max");
+    Assert.assertEquals(1, responseNodeList.size());
+    validatePagingRSSMetric(responseNodeList.get(0));
+
+    //read metric from all nodes in cluster
+    responseNodeList =
+        readMetric(PERFORMANCE_ANALYZER_BASE_ENDPOINT + "/metrics/?metrics=Paging_RSS&agg=max&nodes=all");
+    int nodeNum = nodeIDs.size();
+    Assert.assertEquals(nodeNum, responseNodeList.size());
+    for (int i = 0; i < nodeNum; i++) {
+      validatePagingRSSMetric(responseNodeList.get(i));
+    }
+  }
+
+  /**
+   * check if major page fault is greater or equals to 0. major page fault heavily depends on the workload on OS.
+   * if docker image is running on a OS without heavy workload, we might unlikely observe any major
+   * page fault during the 5s interval.
+   * We might want to revisit this and see if we can run some workload to trigger page fault
+   * {
+   *     "JtlEoRowSI6iNpzpjlbp_Q": {
+   *         "data": {
+   *             "fields": [
+   *                 {
+   *                     "name": "Paging_MajfltRate",
+   *                     "type": "DOUBLE"
+   *                 }
+   *             ],
+   *             "records": [
+   *                 [
+   *                     0.0
+   *                 ]
+   *             ]
+   *         },
+   *         "timestamp": 1606861150000
+   *     }
+   * }
+   */
+  private void validateMajorPageFaultMetric(JsonResponseNode responseNode) throws Exception {
+    Assert.assertTrue(responseNode.getTimestamp() > 0);
+    JsonResponseData responseData = responseNode.getData();
+    Assert.assertEquals(1, responseData.getFieldDimensionSize());
+    Assert.assertEquals(OSMetrics.PAGING_MAJ_FLT_RATE.toString(), responseData.getField(0).getName());
+    Assert.assertEquals(Constants.DOUBLE, responseData.getField(0).getType());
+    Assert.assertEquals(1, responseData.getRecordSize());
+    Assert.assertTrue(responseData.getRecordAsDouble(0, OSMetrics.PAGING_MAJ_FLT_RATE.toString()) >= 0);
```

Comment:
won't the value of `PAGING_MAJ_FLT_RATE ` always be `>=0`?

Reply:
yes, it will always be >= 0 if the page metrics are read correctly. unfortunately this is the only sanity check we can do for this metric because the occurance of page fault is out of our control and it would be hard to write a IT to force OS to trigger page fault